### PR TITLE
Feature: initialise `Connection` without opening a connection and open function exposed

### PR DESCRIPTION
**Pull request by joshstevens19**
Hi guys, love the library just starting integrating with it the create full blown node package for SQL server. You probably will see me expanding on this library a lot. 

The feature i have put in is the ability to create a `Connection` instance without creating a connection straight away. I know there is times when i may want to create the connection instance but not open the connection until i physically have too. I think we should allow people to make that judgement in when there database connection is going to be opened. This ability gives flexibility for people who want to do both.

I wanted this because my library is going to work similar theme of c# `system.data` where you have to physically open the connection when you want too. I made it a optional parameter with default true so it will not break anything for anyone. 

I also want to add an `open` method to your types library (if you have one).

Thanks guys!
Josh

### CODE REVIEW

**Comment by joshstevens19** [comment]
The Travis build is falling over on 2 of the nodes the other 2 pass, it seems its complaining about another package which can not be found (nothing i have done). 

If someone who has access can fix that travis build as well that be awesome! The appveyor integration builds passed.

**Comment by v-suhame** [comment]
As for the types library, there is https://www.npmjs.com/package/@types/tedious, it refers to older version of tedious, we have plans to update it in the future, you're welcome to contribute :)

**Comment by joshstevens19** [comment]
All done and fixed thanks. Can we merge this in to master now..? I have also sent you a personally message to your inbox be brilliant to get a reply.

Thanks

**Comment by Hadis-Fard** [comment]
@joshstevens19 Could you please add unit tests for this?

**Comment by v-suhame** [comment]
+Integration test

**Comment by joshstevens19** [comment]
Course il add some of both tomorrow 👍

**Comment by joshstevens19** [comment]
Hi guys,

I have added integration tests for this in `connection-test.js`. I do not see any unit tests for `connection-test`. In theory the only way to test this properly is through integration tests as it is physically connecting to SQL Server so i see no need to write any unit tests (this is probably why there is none at the moment).

If there is any more tests you want me to add let me know, if not can we merge this in as i want to use this feature :)

Thanks
Josh

**Comment by v-suhame** [comment]
Thanks @joshstevens19! Can you also add the API to [github.io](http://tediousjs.github.io/tedious/api-connection.html) pages?

File: package.json [hunk 1]
```diff
@@ -7,6 +7,7 @@
     "Bryan Ross <bryan@rossipedia.com> (https://github.com/rossipedia)",
     "Ciaran Jessup <ciaranj@gmail.com>",
     "Cort Fritz <cfritz@caa.com>",
+    "Josh Stevens <joshstevens19@hotmail.co.uk> (https://github.com/joshstevens19)",
```
[the remark this answers is not in the document]
Comment (joshstevens19) [reply package.json]: done

**Comment by joshstevens19** [comment]
@v-suhame can you point me to how i can edit that documentation and what the process is for it?

**Comment by joshstevens19** [comment]
seems like the AppVeyor 1 of the nodes has failed again.. you may need to reset it 👍 .. rest built fine,

**Comment by v-suhame** [comment]
tedious uses [github pages](https://guides.github.com/features/pages/) for documentation, you can clone the `gh-pages` branch, add the new API and submit your PR to that branch. 
One way to test changes locally: http://kbroman.org/simple_site/pages/local_test.html

**Comment by joshstevens19** [comment]
Ok il get this sorted tomorrow then hopefully we can merge it in 😂! We will get there in the end don’t you worry ha ha !

File: src/connection.js [hunk 2]
```diff
@@ -68,7 +68,7 @@ function deprecateNonNumberConfigValue(optionName, value) {
 }
 
 class Connection extends EventEmitter {
-  constructor(config) {
```
Comment (arthurschreiber) [inline src/connection.js]: Could this be part of the `config` hash? Like `connectImmediately` or something like that?

File: src/connection.js [hunk 3]
```diff
@@ -488,7 +488,9 @@ class Connection extends EventEmitter {
     this.createTokenStreamParser();
     this.inTransaction = false;
     this.transactionDescriptors = [new Buffer([0, 0, 0, 0, 0, 0, 0, 0])];
-    this.transitionTo(this.STATE.CONNECTING);
+    if (openConnection) {
+      this.open();
```
Comment (arthurschreiber) [inline src/connection.js]: Would you mind moving this to be the very last thing happening inside `constructor`?

**Comment by MichaelSun90** [comment]
Hi @arthurschreiber, This looks like a small feature. We could update the branch and merge this one in. What' s your option on this?
